configs/peer/genesis.json

```diff
@@ -310,182 +266,152 @@
                 "NewParameter": {
-                  "Parameter": "?WSVAssetDefinitionMetadataLimits=1048576,4096_ML",
-                  "source_id": {
-                    "AccountId": "alice@wonderland"
-                  }
+                  "Parameter": "?WSVAssetDefinitionMetadataLimits=1048576,4096_ML"
                 }
               },
               {
                 "NewParameter": {
-                  "Parameter": "?WSVAccountMetadataLimits=1048576,4096_ML",
-                  "source_id": {
-                    "AccountId": "alice@wonderland"
-                  }
+                  "Parameter": "?WSVAccountMetadataLimits=1048576,4096_ML"
                 }
               },
               {
                 "NewParameter": {
-                  "Parameter": "?WSVDomainMetadataLimits=1048576,4096_ML",
-                  "source_id": {
-                    "AccountId": "alice@wonderland"
-                  }
+                  "Parameter": "?WSVDomainMetadataLimits=1048576,4096_ML"
                 }
               },
               {
                 "NewParameter": {
-                  "Parameter": "?WSVIdentLengthLimits=1,128_LL",
-                  "source_id": {
-                    "AccountId": "alice@wonderland"
-                  }
+                  "Parameter": "?WSVIdentLengthLimits=1,128_LL"
                 }
               },
               {
                 "NewParameter": {
-                  "Parameter": "?WASMFuelLimit=1000000",
-                  "source_id": {
-                    "AccountId": "alice@wonderland"
-                  }
+                  "Parameter": "?WASMFuelLimit=1000000"
                 }
               },
               {
                 "NewParameter": {
-                  "Parameter": "?WASMMaxMemory=524288000",
-                  "source_id": {
-                    "AccountId": "alice@wonderland"
-                  }
+                  "Parameter": "?WASMMaxMemory=524288000"
                 }
               }
             ]
           }
+        },
+        {
+          "Register": {
+            "NewRole": {
+              "id": "USER_METADATA_ACCESS",
+              "permissions": [
+                {
+                  "definition_id": "can_remove_key_value_in_user_account",
+                  "params": {
+                    "account_id": {
+                      "AccountId": "alice@wonderland"
+                    }
+                  }
+                },
+                {
+                  "definition_id": "can_set_key_value_in_user_account",
+                  "params": {
+                    "account_id": {
+                      "AccountId": "alice@wonderland"
+                    }
+                  }
+                }
+              ]
+            }
+          }
+        },
+        {
+          "Register": {
+            "Validator": {
+              "id": "permission_validator%genesis@genesis",
+              "validator_type": "Instruction",
+              "wasm": {
```

Comment:
Yes.  A monolithic architecture allows you to define the permission validation with lots of granularity. You can define more stuff and static analysis will enforce the rest of the interactions. @Arjentix was originally on the side of a modular design, where the interactions of the different modules were fixed, but I firmly believe that runtimes should be upgradeable as a whole.

Reply:
Yes, we decided to use one big blob instead of a lot of small ones. However all validators are implemented as a separate crates so that user can import some of them if they need a default implementation we provide